### PR TITLE
Fix/camera null pointer exception

### PR DESCRIPTION
**Pull request by MikeWoots**
By passing the Camera class instance into the CameraEventDelegate it should prevent us from attempting to access a null object and removes the chance of race conditions

### CODE REVIEW

File: android/src/main/java/com/aerobotics/DjiMobile/DJIMobile.java [hunk 1]
```diff
@@ -127,15 +127,17 @@ public void onProductDisconnect() {
...
       @Override
-      public void onComponentChange(BaseProduct.ComponentKey componentKey, BaseComponent baseComponent, BaseComponent baseComponent1) {
-        // TODO
+      public void onComponentChange(BaseProduct.ComponentKey componentKey, BaseComponent oldBaseComponent, BaseComponent newBaseComponent) {
         if (startVisionControlStateListenerOnNextConnect) {
           startVisionControlStateListener();
         }
+
+        if (newBaseComponent != null && newBaseComponent instanceof Camera) {
+          sdkEventHandler.initCameraEventDelegate((Camera) newBaseComponent);
```
Comment (MikeWoots) [inline android/src/main/java/com/aerobotics/DjiMobile/DJIMobile.java:139]: This is the crux of the change @NickHAnderson019 @peza8

**Review by nhanders — APPROVED** [review]
This all looks good to me @MikeWoots. I assume you've done a bunch of testing?

**Comment by MikeWoots** [comment]
@NickHAnderson019 I have tested on the Sim with the debug version. Will test again with the prod version once this is approved

File: android/src/main/java/com/aerobotics/DjiMobile/CameraEventDelegate.java [hunk 2]
```diff
@@ -33,76 +20,31 @@ public void sendEvent(Object payload) {
...
-  CameraEventDelegate() {
-    boolean cameraConnected = isCameraConnected();
-    if (!cameraConnected) {
-      startCameraConnectionListener();
-    } else {
-      isCameraConnected = true;
-      setCameraCallbacks();
-    }
+  CameraEventDelegate(Camera camera) {
+    setCameraCallbacks(camera);
```
Comment (peza8) [inline android/src/main/java/com/aerobotics/DjiMobile/CameraEventDelegate.java:26]: Nice clean code style 👏

Comment (peza8) [reply android/src/main/java/com/aerobotics/DjiMobile/CameraEventDelegate.java:26]: This is infinitely cleaner than what's on the left. Nice one - great "brain wave" 👍

**Review by peza8 — APPROVED** [review]
I don't have context on the minutia, but I'm very much in favour of passing a reference to the camera object around in this flow. 

Would be good to confirm that this reference, which will be a pointer under the syntactic sugar, can't/won't get deallocated by DJI during this operation. But this is probably highly unlikely. 

Approving but keen to see the test results tomorrow 👍

**Pull request merged**